storage/range.go

```diff
@@ -805,10 +841,19 @@ func (r *Range) applyRaftCommand(ctx context.Context, index uint64, originNodeID
 		if !committed {
 			// We didn't commit the batch, but advance the last applied index nonetheless.
 			if err := setAppliedIndex(r.rm.Engine(), r.Desc().RaftID, index); err != nil {
```

Comment:
sorry, I was mistaken. It's most elegant to do this up here, we want to do this rain or shine and there are a bunch of points at which the main function returns (most prominently if `executeCmd` comes back with an error).